src/pages/Admin/Venue/Users/VenueUsersReport.tsx

```diff
@@ -0,0 +1,166 @@
+import React, { useEffect, useCallback, useState } from "react";
+import { useWorldUsers } from "hooks/users";
+import "./VenueUsersReport.scss";
+import Table from "react-bootstrap/Table";
+import Form from "react-bootstrap/Form";
+import { downloadGeneratedCSVFile } from "utils/csv";
+import { User } from "types/User";
+import { WithId } from "utils/id";
+
+const VenueUsersReport: React.FC = () => {
+  const { worldUsers } = useWorldUsers();
+
+  const [headers, setHeaders] = useState<Array<string>>([]);
+  const [headersSelected, setHeadersSelected] = useState<Array<string>>([]);
+  const [customFieldValue, setCustomFieldValue] = useState("");
+
+  const userIsComplete = useCallback((user: WithId<User>) => {
+    return user.partyName && user.pictureUrl;
+  }, []);
+
+  const getUsersHeaders = useCallback(() => {
+    let index = 0;
+    while (index < worldUsers.length) {
+      if (userIsComplete(worldUsers[index])) {
+        setHeaders(Object.keys(worldUsers[index]));
+        return;
+      } else {
+        index++;
+      }
+    }
+  }, [worldUsers, userIsComplete]);
+
+  useEffect(() => {
+    getUsersHeaders();
+  }, [worldUsers, getUsersHeaders]);
+
+  const renderPageHeader = () => {
+    return <h4>Users Reports</h4>;
+  };
+
+  const renderUserTable = () => {
+    return (
+      <Table striped bordered hover>
+        <thead>
+          <tr>
+            <th>#</th>
+            <th>Party Name</th>
+            <th>Full Name</th>
+            <th>Department</th>
+          </tr>
+        </thead>
+        <tbody>
+          {/* when I try to use WithId<User> as realName are not added in it, it shows erro, so that's why I need any */}
+          {/* eslint-disable-next-line*/}
+          {worldUsers.map((user: any, index) => {
+            return (
+              userIsComplete(user) && (
+                <tr key={index}>
+                  <td>{index + 1}</td>
+                  <td>{user.partyName || "Not"}</td>
+                  <td>{user.realName || "Not set"}</td>
+                  <td>{user.companyDepartment || "Not set"}</td>
+                </tr>
+              )
+            );
+          })}
+        </tbody>
+      </Table>
+    );
+  };
+
+  const downloadCSV = useCallback(() => {
+    downloadGeneratedCSVFile(worldUsers, headersSelected);
+  }, [worldUsers, headersSelected]);
+
+  const renderDownloadBtn = () => {
+    return (
+      <button
+        className="btn btn-primary download-btn"
+        onClick={() => downloadCSV()}
+      >
+        Download{" "}
+      </button>
+    );
+  };
+
+  const updateSelectedHeaders = useCallback(
+    (selected: string) => {
+      if (headersSelected.includes(selected)) {
+        headersSelected.splice(headersSelected.indexOf(selected), 1);
+        setHeadersSelected([...headersSelected]);
+      } else {
+        setHeadersSelected([...headersSelected, selected]);
+      }
+    },
+    [headersSelected]
+  );
```

Comment:
What's the intent of this code?

Reply:
to select which fields they want to export, this func will update the selection